migration-guide.adoc

```diff
@@ -58,6 +58,46 @@ Multiple component mappings for the same java class with different property mapp
 
 Passing null or not is now triggered by whether setting the parameter was called at all.  In other ords a distinction is made between calling `setParameter` passing null versus not calling `setParameter` at all.  In the first case, we pass along the null value; in the second we do not.
 
+=== Query
+
+=== HQL results
+
+Hql queries that joins an Entity `Person` with an Entity `Address` without specifying a select clause `from Address a join a.address` does not return anymore a `List<Object[]>`but a list of `Person`.
+
+The hql query `select a,ad from Address a join a.address ad` returns instead a `List<Object[]>`.
+
+e.g.
+```
+@Entity
+class Person {
+    ...
+
+    @ManyToOne
+    Address address
+
+    ...
+}
+
+@Entity
+class Address {
+
+    ...
+
+}
+
+List<A> result = session.createQuery("from Address a join a.address").list()
+
+```
+
+
+==== Stream
+
+`jakarta.persistence.Query#getResultStream()` and `org.hibernate.query.Query#stream()` do not return anymore a Stream decorator so in order to close the underlying Stream is now necessary to explicitly call the `Stream#close()` method.
```

Comment:
```suggestion
`jakarta.persistence.Query#getResultStream()` and `org.hibernate.query.Query#stream()` do not return a `Stream` decorator anymore, so in order to close the underlying JDBC `ResultSet` it is now necessary to explicitly call the `Stream#close()` method. The JDK `Stream` documentation is quite explicit about the need for an explicit call to `close` by the user to avoid resource leakages, so we build upon this requirement.
```

Reply:
Change made, thanks @beikov